Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -46,18 +46,10 @@ namespace AzToolsFramework
             m_prefabPublicRequestHandler.Connect();
             m_prefabSystemScriptingHandler.Connect(this);
             AZ::SystemTickBus::Handler::BusConnect();
-            if (AzToolsFramework::Prefab::IsHotReloadingEnabled())
-            {
-                AzToolsFramework::AssetSystemBus::Handler::BusConnect();
-            }
         }
 
         void PrefabSystemComponent::Deactivate()
         {
-            if (AzToolsFramework::Prefab::IsHotReloadingEnabled())
```

Comment:
It keeps the bus handler always active to listen to source file deletions. But we are enabling the hot reloading only when the deleted source file is a prefab and there are instances of it in the current level. So, rather than opening the floodgates to detect every prefab change on file, we will be carefully enabling hot reloading for only those cases that are causing issues to users.